### PR TITLE
WIP on setting draft state

### PR DESCRIPTION
**Pull request by ECruz49**
https://www.pivotaltracker.com/story/show/158241353

### CODE REVIEW

File: src/pages/AddProperty/AddProp.jsx [hunk 1]
```diff
@@ -123,7 +124,9 @@ class AddProp extends React.Component {
             <div>
                 <TopNav />
                 <div className="saveDraft-publish-buttons">
-                    <button>Save As Draft</button>
+                    <button
+                        onClick={() => this.setState({draft: true})}>
```
Comment (BrianJenney) [inline src/pages/AddProperty/AddProp.jsx]: I think the easiest way is to call the `submitProperty = (bool) => {`... function and pass in either true or false depending on which button was clicked.

Then `draft: bool`

Comment (BrianJenney) [reply src/pages/AddProperty/AddProp.jsx]: Lemme know if you have any questions on that

File: src/pages/AddProperty/AddProp.jsx [hunk 2]
```diff
@@ -125,8 +126,8 @@ class AddProp extends React.Component {
                 <TopNav />
                 <div className="saveDraft-publish-buttons">
                     <button
-                        onClick={() => this.setState({draft: true})}>
-                        Save As Draft</button>
+                        onClick={this.submitProperty(true)}
+                    >Save As Draft</button>
                     <button id="pubButton">Publish Listing</button>
```
Comment (BrianJenney) [inline src/pages/AddProperty/AddProp.jsx]: Attach that same function to the publish button and set it to false. When you click the button does it work as expected? No errors in the console?

Comment (BrianJenney) [inline src/pages/AddProperty/AddProp.jsx]: Attach that same function to the publish button and set it to false. When you click the button does it work as expected? No errors in the console?

Comment (ECruz49) [reply src/pages/AddProperty/AddProp.jsx]: Nothing happens. Although, when the page reloads, it gives me an alert to add more pictures.

Comment (BrianJenney) [reply src/pages/AddProperty/AddProp.jsx]: @ECruz49 I think you need to do `this.submitProperty.bind(true)` in order to pass the bool to the function or else the function invokes when the page is loaded which is why you are getting that alert. Try it out and maybe console.log the results in the function to see if it is working as expected

**Comment by ECruz49** [comment]
Updated file to use this.submitProperty.bind(true) for the onClick function. 

Tested the 'Save as Draft' button earlier, and the page would take me to the home page. Tested later today, nothing happens except for a Network Error with POST pointing to 119 in AddProp.jsx. Still WIP

File: src/pages/AddProperty/AddProp.jsx [hunk 3]
```diff
@@ -45,7 +45,8 @@ class AddProp extends React.Component {
         sqFeet: 0,
         sqFeetLotSize: 0,
         yearBuilt: Date.now(),
-        form: new FormData()
+        form: new FormData(), 
+        draft: false
```
Comment (BrianJenney) [inline src/pages/AddProperty/AddProp.jsx]: this should actually be `status` and we will set it to either `publish` or `draft`

File: src/pages/AddProperty/AddProp.jsx [hunk 4]
```diff
@@ -89,7 +90,7 @@ class AddProp extends React.Component {
         this.setState(propertyInfo);
     };
 
-    submitProperty = () => {
+    submitProperty = bool => {
```
Comment (BrianJenney) [inline src/pages/AddProperty/AddProp.jsx]: change to `status`

File: src/pages/AddProperty/AddProp.jsx [hunk 5]
```diff
@@ -108,6 +109,7 @@ class AddProp extends React.Component {
         this.state.form.append("yearBuilt", this.state.yearBuilt);
         this.state.form.append("sqFeetLot", this.state.sqFeetLotSize);
         this.state.form.append("sqFeet", this.state.sqFeet);
+        this.state.form.append("draft", bool);
```
Comment (BrianJenney) [inline src/pages/AddProperty/AddProp.jsx]: `this.state.form.append("status", status);`

File: src/pages/AddProperty/AddProp.jsx [hunk 6]
```diff
@@ -123,8 +125,12 @@ class AddProp extends React.Component {
...
                 <TopNav />
                 <div className="saveDraft-publish-buttons">
-                    <button>Save As Draft</button>
-                    <button id="pubButton">Publish Listing</button>
+                    <button
+                        onClick={this.submitProperty.bind(true)}
+                    >Save As Draft</button>
+                    <button id="pubButton"
+                        onClick={this.submitProperty.bind(false)}
+                    >Publish Listing</button>
```
Comment (BrianJenney) [inline src/pages/AddProperty/AddProp.jsx]: `bind(this, 'publish')` or `draft`

File: src/pages/AddProperty/AddProp.jsx (hunk 3 above)
Comment (BrianJenney) [reply src/pages/AddProperty/AddProp.jsx]: This was my bad: I didn't update the backend correctly. Make sure you pull the latest down for the backend to see if this works correctly. I did it on my machine and it works but I wanted to save the work for you 😉  here's some docs about event handling in react: https://reactjs.org/docs/handling-events.html#passing-arguments-to-event-handlers

**Pull request merged**